core/src/main/java/by/ilyin/core/controller/AuthCoreController.java

```diff
@@ -0,0 +1,48 @@
+package by.ilyin.core.controller;
+
+import by.ilyin.core.dto.SignInDTO;
+import by.ilyin.core.entity.CustomJWT;
+import by.ilyin.core.entity.CustomUser;
+import by.ilyin.core.service.auth.AuthService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Set;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api")
+public class AuthCoreController {
+
+    private final AuthService authService;
+
+    @PostMapping("/sign-in")
+    public CustomUser signIn(@RequestBody SignInDTO signInDTO) {
+        return authService.getUserByCredentials(signInDTO);
+    }
+
+    @PostMapping("/logout")
+    public Set<CustomJWT> logout(@RequestParam(value = "userId") Long userId) {
+        return authService.logoutProcess(userId);
+    }
+
+    @GetMapping("/jwt")
```

Comment:
@YauhenIlyin where did you find it in the ticket?

Reply:
we talked with Egor that we use tokens only on the web so far, and how annoying it is to work with the core, he clarifies with you.

but because we have a logout and a blacklist that should be stored in the database, I had to write logic for core.

thus, the validation of tokens takes place on the web now, and inside we use ordinary requests between modules.

Reply:
The task does not specify how to implement logout.
At that moment, I had no one to ask.


since we have a logout,
so there must be a mechanism to make the token invalid before the time expires.
it means we have to store somewhere tokens that have not expired, but are already "logged out".
so we must have a black list of such tokens.